### PR TITLE
Fix tslint path regexp and issues

### LINKED ISSUE #50: tslint doesn't run for './src/app/editor' folder
`"lint": "npm run tslint src/**/*.ts"` in `package.json` must be changed to cover  `./src/app/editor/` folder.

### PR DESCRIPTION
**Pull request by harunurhan**
* Change tslint path to cover all ts files. (fix #50)

* Fix all tslint errors.

* Change tslint config to allow string-literal property access.

* Add new-line to at the end of non *.ts files.

### CODE REVIEW

**Comment by kaplun** [comment]
@harunurhan to ease code review, in general, can you make one commit per feature/bug fix?
Also in the commit log, please follow Invenio conventions:
* http://invenio.readthedocs.io/en/latest/processes/developing.html#code-contributions
* http://invenio.readthedocs.io/en/latest/technology/git.html#r2-remarks-on-commit-log-messages

Namely:
* Do not use verbs in the first line of the commit log
* Decline each bullet line in 3rd person singular. E.g. Fixes... Changes...
* Add prefixes such as NEW BETTER etc. as described in the second above link 

(this will allow in the future to have consistent release notes being automatically generated)

Other than this I trust this PR is doing what you say :dancer: 
Maybe @eamonnmag can have a look? Or @drjova?

**Comment by harunurhan** [comment]
@kaplun I will make sure next PRs will follow Invenio conventions. 

I fixed commit message although I disagree with some of these conventions 😄 

Btw, this PR doesn't change any logic but adds/removes `spaces, commas, semicolons` to fix styling issues, and (almost) all are for `tslint`. That's why I kept it in a single commit.

**Comment by kaplun** [comment]
@harunurhan thanks! Well those conventions have a value (a bit complex to explain here. I'll show you IRL when I am back).

File: src/app/editor/add-field-dropdown/add-field-to-list-dropdown.component.ts [hunk 1]
```diff
@@ -58,7 +58,8 @@ export class AddFieldToListDropdownComponent extends AbstractAddFieldDropdownCom
     let emptyValue = this.emptyValueService.generateEmptyValue(fieldSchema);
     this.values.forEach(value => {
       if (typeof emptyValue === 'object') {
-        // Object.assign is used to create new reference per value, so that they won't be binded to same variable
+        // Object.assign is used to create new reference per value, 
+        // so that they won't be binded to same variable
```
Comment (jacquerie) [inline src/app/editor/add-field-dropdown/add-field-to-list-dropdown.component.ts]: FYI it's *bound*

File: src/app/editor/editor.component.ts [hunk 2]
```diff
@@ -101,6 +101,19 @@ export class EditorComponent extends AbstractTrackerComponent {
...
+    if (validationUrl) {
+      // TODO: change it post and include record and schema in request when it is not mock.
+      this.http.get(validationUrl).map(res => res.json())
+        .subscribe(validationErrors => {
+          this.appGlobalsService.globalErrors = validationErrors;
+        });
+    }
+
+    // TODO: implement saving and do it when only validationErrors is empty.
+  }
```
Comment (jacquerie) [inline src/app/editor/editor.component.ts]: This looks unrelated to the rest of the commit.

**Pull request merged**